aopy/precondition.py

```diff
@@ -1,2 +1,225 @@
 # precondition.py
-# code for preconditioning neural data
\ No newline at end of file
+# code for preconditioning neural data
+from scipy import signal
+from scipy.signal import butter, lfilter, filtfilt, windows, freqz
+import math
+import numpy as np
+import os
+import matplotlib.pyplot as plt
+#
+import nitime.algorithms as tsa
+import nitime.utils as utils
+from nitime.viz import winspect
+from nitime.viz import plot_spectral_estimate
+'''
+Filter functions
+'''
+def generate_test_signal(T, fs, freq, a=0.02):
+    '''
+
+    Args:
+        T (float): Time period in seconds
+        fs (int): Sampling frequency in Hz
+        freq (1D array): Frequencies to be mixed in the test signal. main frequency in the first element
+        plot_true (bool): boolean value if true, plots the test signal in the provided axes
+        ax (plot axis handle): axis to plot the noisy test signal
+
+    Returns:
+        x (1D array): cosine wave with multiple frequencies and noise
+    '''
+    nsamples = int(T * fs)
+    t = np.linspace(0, T, nsamples, endpoint=False)
+    # a = 0.02
+    f0 = freq[0]
+    # noise_power = 0.001 * fs / 2
+
+    x = a * 2 * np.cos(2 * np.pi * f0 * t)  # adding primary frequency
+    x += a * 0.5 * np.cos(2 * np.pi * freq[1] * t)
+    x += a * 1.5 * np.cos(2 * np.pi * freq[2] * t)
+    x += a * 20 * np.sin(2 * np.pi * 1.2 * np.sqrt(t))  # noise
+    # x += np.random.normal(scale=np.sqrt(noise_power), size=t.shape)  # noise
+
+    return x, t, f0
+
+def bandpass_butterworth_params(cutoff_low, cutoff_high, fs, order = 4, filter_type = 'bandpass'):
+    '''
+    Args:
+        cutoff_low (int): lower cut-off frequency (in Hz)
+        cutoff_high (int): higher cutoff frequency (in Hz)
+        fs (int): sampling rate (in Hz)
+        order (int): Order of the butter worth filter
+        filter_type (str) : Type of filter. Accepts one of the four values - {‘lowpass’, ‘highpass’, ‘bandpass’, ‘bandstop’}
```

Comment:
If the function is only designed to create bandpass params, filter_type doesn't need to be in the inputs.

Reply:
removed bandpass from function names

Reply:
i think your solution doesn't really make sense because you're always including cutoff_low and cutoff_high in the filter.

i feel like this is an instance where having a documentation page for filtering that explains how to use scipy.signal.butter would be better than writing this wrapper function